autowarefoundation/autoware
# [Feature] waypoint replanner and extractor

## Status
**DEVELOPMENT**

## Description
Decompose function of /waypoint_loader into the following nodes. autowarefoundation/autoware_ai#274
(load waypoint.csv & publish topic) => /waypoint_loader
(pub/sub topic & replan waypoint) => /waypoin_replanner >>> waypoint_loader.launch
(subscribe topic & save waypoint.csv) => /waypoint_extractor >>> waypoint_saver.launch

![image](https://user-images.githubusercontent.com/33311630/52145569-a9656f80-26a4-11e9-80c6-b9f454685026.png)
![image](https://user-images.githubusercontent.com/33311630/52214772-ce8cf480-28d5-11e9-9879-25825fb8b2d6.png)

## Todos

## Steps to Test or Reproduce
How to run these nodes: [README](https://github.com/CPFL/Autoware/tree/feature/devide_waypoint_loader/ros/src/computing/planning/motion/packages/waypoint_maker)
* waypoint_replanner
 Make sure that replan can be done as before.
* waypoint_extractor
 Confirm that the csv-file is generated.
